<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <title>The source code</title>
  <link href="../resources/prettify/prettify.css" type="text/css" rel="stylesheet" />
  <script type="text/javascript" src="../resources/prettify/prettify.js"></script>
  <style type="text/css">
    .highlight { display: block; background-color: #ddd; }
  </style>
  <script type="text/javascript">
    function highlight() {
      document.getElementById(location.hash.replace(/#/, "")).className = "highlight";
    }
  </script>
</head>
<body onload="prettyPrint(); highlight();">
  <pre class="prettyprint lang-js"><span id='java-lang-reflect-UndeclaredThrowableException'>/**@class java.lang.reflect.UndeclaredThrowableException
</span>@extends java.lang.RuntimeException

 Thrown by a method invocation on a proxy instance if its invocation
 handler's {@link java.lang.reflect.InvocationHandler#invoke invoke} method throws a
 checked exception (a {@code Throwable} that is not assignable
 to {@code RuntimeException} or {@code Error}) that
 is not assignable to any of the exception types declared in the
 {@code throws} clause of the method that was invoked on the
 proxy instance and dispatched to the invocation handler.

 &lt;p&gt;An {@code UndeclaredThrowableException} instance contains
 the undeclared checked exception that was thrown by the invocation
 handler, and it can be retrieved with the
 {@code getUndeclaredThrowable()} method.
 {@code UndeclaredThrowableException} extends
 {@code RuntimeException}, so it is an unchecked exception
 that wraps a checked exception.

 &lt;p&gt;As of release 1.4, this exception has been retrofitted to
 conform to the general purpose exception-chaining mechanism.  The
 &quot;undeclared checked exception that was thrown by the invocation
 handler&quot; that may be provided at construction time and accessed via
 the {@link #getUndeclaredThrowable}() method is now known as the
 &lt;i&gt;cause&lt;/i&gt;, and may be accessed via the {@link Throwable#getCause()} method, as well as the aforementioned &quot;legacy
 method.&quot;

 @author      Peter Jones
 @see         InvocationHandler
 @since       1.3
*/
var UndeclaredThrowableException = {

<span id='java-lang-reflect-UndeclaredThrowableException-method-getUndeclaredThrowable'>/**Returns the {@code Throwable} instance wrapped in this
</span> {@code UndeclaredThrowableException}, which may be {@code null}.

 &lt;p&gt;This method predates the general-purpose exception chaining facility.
 The {@link Throwable#getCause()} method is now the preferred means of
 obtaining this information.
@return {Object {java.lang.Throwable}} the undeclared checked exception that was thrown
*/
getUndeclaredThrowable : function(  ) {},

<span id='java-lang-reflect-UndeclaredThrowableException-method-getCause'>/**Returns the cause of this exception (the {@code Throwable}
</span> instance wrapped in this {@code UndeclaredThrowableException},
 which may be {@code null}).
@return {Object {java.lang.Throwable}} the cause of this exception.
@since 1.4
*/
getCause : function(  ) {},


};</pre>
</body>
</html>
